src/app/modules/test-navigator/test-navigator.component.ts

```diff
@@ -356,4 +358,25 @@ export class TestNavigatorComponent implements OnInit, OnDestroy {
     }
   }
 
+  async onDeleteConfirm(nodeToDelete: TestNavigatorTreeNode): Promise<void> {
+    try {
+      const result = await this.persistenceService.deleteResource(nodeToDelete.id);
```

Comment:
MINOR: we should mark this node as currently being deleted and show that visually (e.g. provide css that makes this file display color gray). This could however be a new ticket.